src/ssl.c

```diff
@@ -5903,7 +5931,7 @@ int AddCA(WOLFSSL_CERT_MANAGER* cm, DerBuffer** pDer, int type, int verify)
 #endif
 
     /* check CA key size */
-    if (verify) {
+    if (verify && (ret == 0 )) {
```

Comment:
Nice find here! Doesn't cause any harm as it was, but can cause the original error code to be overridden.